File: Data Structures & Algorithms/three-integer-sum/submission-6.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()
        
        i = 0
        while nums[i] <= 0 and i < len(nums) - 2:
            j = i + 1
            k = len(nums) - 1
            while j < k:
                if nums[j] + nums[k] == -nums[i]:
                    res.append([nums[i], nums[j], nums[k]])
                    j += 1
                    k -= 1
                    while j < len(nums) and nums[j] == nums[j - 1]:
                        j += 1
                    while k > j and nums[k] == nums[k + 1]:
                        k -= 1
                elif nums[j] + nums[k] > -nums[i]:
                    k -= 1
                else:
                    j += 1
            i += 1
            while i < len(nums) - 2 and nums[i] == nums[i - 1]:
                i += 1
        return res
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-6.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i : nums) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }

        PriorityQueue<Integer> pq = new PriorityQueue<>((a, b) -> {
            return map.get(b) - map.get(a);
        });

        for (Map.Entry<Integer, Integer> e : map.entrySet()) {
            pq.offer(e.getKey());
        }

        int[] soln = new int[k];
        for (int i = 0; i < k; i++) {
            soln[i] = pq.poll();
        }
        return soln;
    }
}
